osu.Game.Resources/Shaders/sh_ArgonBarPathUtils.h

```diff
@@ -0,0 +1,198 @@
+    highp vec2 progressRange,
+    mediump vec2 arc1End,
+    mediump vec2 arc2Start,
+    highp float progressArc1End,
+    highp float progressArc2Start,
+    mediump vec2 pixelAbsolutePos)
+{
+    if (progressRange.x > progressArc2Start || progressRange.y < progressArc1End)
+        return 1000.0;
+
+    progressRange.x = max(progressRange.x, progressArc1End);
+    progressRange.y = min(progressRange.y, progressArc2Start);
+
+    mediump vec2 startPos = mix(arc1End, arc2Start, (progressRange.x - progressArc1End) / (progressArc2Start - progressArc1End));
+    mediump vec2 endPos = mix(arc1End, arc2Start, (progressRange.y - progressArc1End) / (progressArc2Start - progressArc1End));
+    
+    return dstToLine(startPos, endPos, pixelAbsolutePos);
+}
+
+highp float dstToBottomArc(
+    highp vec2 progressRange,
+    mediump float radius,
+    highp float progressArc2Start,
+    highp float progressArc2End,
+    mediump vec2 arcCentre,
+    highp float slashAngle,
+    highp float fullArcAngle,
+    mediump vec2 pixelAbsolutePos)
+{
+    if (progressRange.x > progressArc2End || progressRange.y < progressArc2Start)
+        return 1000.0;
+
+    progressRange.x = max(progressRange.x, progressArc2Start);
+    progressRange.y = min(progressRange.y, progressArc2End);
+
+    highp float currentArcAngle = fullArcAngle * (progressRange.y - progressRange.x) / (progressArc2End - progressArc2Start);
+    highp float rotationOffset = fullArcAngle * (progressRange.x - progressArc2Start) / (progressArc2End - progressArc2Start);
+    return dstToArc(arcCentre, radius, currentArcAngle, pixelAbsolutePos, PI + HALF_PI + (slashAngle - fullArcAngle + currentArcAngle) * 0.5 + rotationOffset);
+}
+
+highp float dstToBottomLine(
+    highp vec2 progressRange,
+    mediump vec2 arc2End,
+    mediump vec2 p4,
+    highp float progressArc2End,
+    mediump vec2 pixelAbsolutePos)
+{
+    if (progressRange.y < progressArc2End)
+        return 1000.0;
+
+    progressRange.x = max(progressRange.x, progressArc2End);
+
+    mediump vec2 startPos = mix(arc2End, p4, (progressRange.x - progressArc2End) / (1.0 - progressArc2End));
+    mediump vec2 endPos = mix(arc2End, p4, (progressRange.y - progressArc2End) / (1.0 - progressArc2End));
+    
+    return dstToLine(startPos, endPos, pixelAbsolutePos);
+}
+
+highp float getBarTexturePosition(mediump vec2 size, highp vec2 progressRange, mediump float padding, mediump vec2 pixelAbsolutePos)
+{
+    mediump vec2 p1 = vec2(min(padding, size.x * 0.5), min(padding, size.y * 0.5));
+    mediump vec2 p4 = vec2(max(size.x - padding, size.x * 0.5), max(size.y - padding, size.y * 0.5));
+
+    if (p4.y == p1.y)
+    {
+        return dstToLine(vec2(p1.x + progressRange.x * (p4.x - p1.x), p1.y), vec2(p1.x + progressRange.y * (p4.x - p1.x), p1.y), pixelAbsolutePos);
+    }
+
+    mediump float topWidth = max(size.x - padding - 70.0, padding) - p1.x;
+    mediump float bottomWidth = p4.x - max(size.x - padding - 40.0, padding);
```

Comment:
Can these numbers exist as constant members at the top of the header file? If I'm understanding their purpose, they're defining the points at which the curve from the top to the bottom begins.